### ISSUE TITLE
4.Raport structurat în consolă pe secțiuni. Ștefania(Gr.B)

### ISSUE THREAD

**Issue by licastefania200001-alt** [georgerusu017/churn_rate_predicition#52]

**Comment by licastefania200001-alt** [comment]
import pandas as pd
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, average_precision_score

# 1. Citim fisierul CSV
df = pd.read_csv("tranzactii.csv", sep='\t')

# 2. Valorile reale (etichetele)
y_true = df['Class'].astype(int).tolist()  # 1 = frauda, 0 = normal

# 3. Exemplu de scor: suma unor coloane relevante
y_score = df[['f1','f2','f3','f4','f5']].sum(axis=1).tolist()

# 4. Alegerea pragului
threshold = 0.5
y_pred = [1 if s >= threshold else 0 for s in y_score]

# 5. Calculul matricii de confuzie
cm = confusion_matrix(y_true, y_pred)

# 6. Calculul metricilor
accuracy = accuracy_score(y_true, y_pred)
precision = precision_score(y_true, y_pred, zero_division=0)
recall = recall_score(y_true, y_pred, zero_division=0)
pr_auc = average_precision_score(y_true, y_score)

# 7. Pregatirea raportului
raport = []
raport.append("===== RAPORT DE ANALIZA A TRANZACTIILOR =====\n")
raport.append("--- Informatii generale ---")
raport.append(f"Număr total de tranzacții: {len(y_true)}")
raport.append(f"Pragul ales pentru clasificare: {threshold}\n")

raport.append("Matrice de confuzie")
raport.append(str(cm) + "\n")

raport.append(" Metrici")
raport.append(f"Accuracy : {accuracy:.4f}")
raport.append(f"Precision: {precision:.4f}")
raport.append(f"Recall   : {recall:.4f}")
raport.append(f"PR-AUC  : {pr_auc:.4f}\n")

raport.append("Observatii")
if sum(y_true) < 5:
    raport.append("Clasa pozitiva are sub 5 cazuri, metricile pot fi instabile.")
else:
    raport.append("Distributia claselor pare suficienta pentru o analiza simpla.")
raport.append("PR-AUC este calculat pe scorul agregat al caracteristicilor f1–f5.")
raport.append("Interpretare: valori mai mari indica o performanta mai buna a detectarii fraudelor.\n")
raport.append("====================================================\n")

# 8. Salvarea raportului intr-un fisier
with open("raport_tranzactii.txt", "w") as f:
    for linie in raport:
        f.write(linie + "\n")

print("Raportul a fost salvat in 'raport_tranzactii.txt'.")
